runners/apex/src/main/java/org/apache/beam/runners/apex/translation/operators/ApexReadUnboundedInputOperator.java

```diff
@@ -106,6 +104,18 @@ private void emitWatermarkIfNecessary(long mark) {
 
   @Override
   public void endWindow() {
+    if (outputWatermark == GlobalWindow.TIMESTAMP_MAX_VALUE.getMillis()) {
```

Comment:
This should really be `BoundedWindow.TIMESTAMP_MAX_VALUE`. It is inheritance on a static field that isn't really meant for inheritance (but, Java).

And even though this is fine, I would feel slightly better by using `>=` (aka `.isAfter`) which lets the reader know the idea and could be more lenient if some variation occurs accidentally.